ADC/Core/Src/main.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * 	Implement a routine that reads the internal voltage reference (ADC_IN17)
+ * 	and internal temperature sensor (ADC_IN18).
+ * 	The internal reference voltage must be in millivolts and the temperature in Celsius.
+ *
+ * 	1 - Polling		[/]
+ * 	2 - Interrupt	[ ]
+ * 	3 - DMA			[ ]
+ */
+
+#include "stm32l053xx.h"
+#include "macros.h"
+#include "init.h"
+
+uint8_t		ADC_error	= 0;	//error management flag
+uint16_t	measure 	= 0,	//AD read after a conversion
+			v_ref 		= 0;	//internal reference voltage measured (after calculation)
+int16_t		temperature	= 0;	//temperature in Celsius degrees measured (after calculation)
+
+void ADC_Config (uint8_t Channel);
+void Wait	 (uint16_t);
+
+int main (void)
+{
+	ADC_Init();
+
+	while(1)
+	{
+		ADC_Config(CH_INT_TEMP);			//internal temperature reading
+		ADC1->CR |= ADC_CR_ADSTART;
+		while(!(ADC1->ISR & ADC_ISR_EOC));
+		measure = ADC1->DR & ADC_DR_DATA;
+
```

Comment:
Why do you have the expression `ADC1->DR & ADC_DR_DATA` to read the ADC data?

Reply:
what do you mean? I just store the data read in measure variable to use it for the calculations right after. Should I use the register instead of the variable then?